pkg/operation/botanist/etcdencryption.go

```diff
@@ -0,0 +1,332 @@
+	return nil
+}
+
+// RewriteShootSecretsIfEncryptionConfigurationChanged rewrites a shoot's secrets if the EncryptionConfiguration has changed
+func (b *Botanist) RewriteShootSecretsIfEncryptionConfigurationChanged() error {
+	b.Logger.Info("starting RewriteShootSecretsIfEncryptionConfigurationChanged")
+	// WARNING:
+	// No explicit checking of whether EncryptionConfiguration is contained in a backup.
+	// Be aware of the risk!
+	//
+	// TODO: Ensure this is also agreed upon by Gardener team
+
+	// TODO: contact Amshuman Rao Karaya
+
+	enableEtcdEncryption, err := utils.CheckVersionMeetsConstraint(b.Shoot.Info.Spec.Kubernetes.Version, ">= 1.13")
+	if err != nil {
+		return err
+	}
+	if !enableEtcdEncryption {
+		return nil
+	}
+	needToRewrite, err := b.needToRewriteShootSecrets()
+	if err != nil {
+		return err
+	}
+	if needToRewrite {
+		b.Logger.Info("rewriting shoot secrets")
+		err = b.rewriteShootSecrets()
+		if err != nil {
+			return err
+		}
+	} else {
+		b.Logger.Trace("EncryptionConfiguration unchanged, no rewriting necessary")
+	}
+	return nil
+}
+
+// readEncryptionConfigurationFromSeed reads the EncryptionConfiguration from the shoot namespace in the seed
+func (b *Botanist) readEncryptionConfigurationFromSeed() (exists bool, ec *apiserverconfigv1.EncryptionConfiguration, forcePlaintext bool, err error) {
+	client := b.Operation.K8sSeedClient
+	ecs, err := client.GetSecret(b.Operation.Shoot.SeedNamespace, common.EtcdEncryptionSecretName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil, false, nil
+		} else {
+			return false, nil, false, err
+		}
+	}
+	secretData, ok := ecs.Data[common.EtcdEncryptionSecretFileName]
+	if !ok {
+		return true, nil, false, fmt.Errorf("EncryptionConfiguration in seed cluster (%v) does not contain expected element: %v", common.EtcdEncryptionSecretName, common.EtcdEncryptionSecretFileName)
+	}
+	ec, err = encryptionconfiguration.CreateFromYAML(secretData)
+	if err != nil {
+		return true, nil, false, fmt.Errorf("EncryptionConfiguration in seed cluster (%v) is not consistent: %v", common.EtcdEncryptionSecretName, err)
+	}
+	_, ok = ecs.Annotations[common.EtcdEncryptionForcePlaintextAnnotationName]
+	if !ok {
+		return true, ec, false, nil
+	}
+	return true, ec, true, nil
+}
+
+func (b *Botanist) calculateEtcdEncryptionSecretNameInGardenCluster() string {
+	secretName := fmt.Sprintf("%s.%s", b.Shoot.Info.Name, common.EtcdEncryptionSecretName)
+	return secretName
+}
+
+// readEncryptionConfigurationFromGarden reads the EncryptionConfiguration from the shoot namespace in the seed
+func (b *Botanist) readEncryptionConfigurationFromGarden() (bool, *apiserverconfigv1.EncryptionConfiguration, error) {
+	client := b.Operation.K8sGardenClient
+	secretNameInGardenCluster := b.calculateEtcdEncryptionSecretNameInGardenCluster()
+	ecs, err := client.GetSecret(b.Shoot.Info.Namespace, secretNameInGardenCluster)
```

Comment:
```suggestion
	ecs, err := b.K8sGardenClient.GetSecret(b.Shoot.Info.Namespace, secretNameInGardenCluster)
```

and remove `client := b.Operation...`